packages/appinio_swiper/lib/appinio_swiper.dart

```diff
@@ -1,679 +1,743 @@
+import 'dart:collection';
 import 'dart:math';
 
 import 'package:appinio_swiper/types.dart';
 import 'package:flutter/material.dart';
 
-import 'controllers.dart';
 import 'enums.dart';
 
-export 'controllers.dart';
 export 'enums.dart';
 export 'types.dart';
 
 class AppinioSwiper extends StatefulWidget {
-  /// when the user cancels the swipe
-
+  /// A callback that's called when the user lifts their finger during a drag
+  /// without having reached [threshold].
   final VoidCallback? onSwipeCancelled;
 
-  /// widget builder for creating cards
-  final CardsBuilder cardsBuilder;
+  /// The indexed widget builder that builds a card for the given index.
+  final IndexedWidgetBuilder cardsBuilder;
 
-  ///cards count
-  final int cardsCount;
+  /// The number of cards in the stack.
+  final int cardCount;
```

Comment:
Convention is to use singular form for variables suffixed in count. If you guys have strong feelings about this I can roll it back.